chore: allowedLateness doc and fix flaky test (#708)
numaproj/numaflow: docs/user-guide/user-defined-functions/reduce/reduce.md

This applies to all the window types regardless of whether they are keyed or non-keyed windows. 

## Allowed Lateness

> By default, a late message will be dropped if the window has already been closed, but sometimes user may want to accept
> late messages. Therefore, we support the `allowedLateness` feature for the reduce operations to allow late data to be 
> processed by slowing the down the close-of-book operation of the desired reduce vertex. Late data will be included for
> the reduce operation as long as the late data is not later than `(CurrentWatermark - AllowedLateness)`. Each reduce 
> vertex can have its own `allowedLateness`.

Reviewer: ```suggestion
`allowedLateness` flag on the Reduce vertex that will allow late data to be 
processed by slowing the down the close-of-book operation of the Reduce vertex. Late data will be included for
the Reduce operation as long as the late data is not later than `(CurrentWatermark - AllowedLateness)`. Without `allowedLateness`, late data will be rejected and will not be included in the Reduce operation. Each Reduce 
vertex can have its own `allowedLateness`.
```

```yaml
vertices:
  - name: my-udf
    udf:
      groupBy:

>         allowedLateness: 1s # Optional, allowedLateness is disabled by default

Reviewer: ```suggestion
        allowedLateness: 5s # Optional, allowedLateness is disabled by default
```
